app/src/main/java/com/github/se/icebreakrr/ui/authentication/ProfileCreation.kt

```diff
@@ -0,0 +1,314 @@
+import android.app.DatePickerDialog
+import android.widget.Toast
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Check
+import androidx.compose.material.icons.filled.DateRange
+import androidx.compose.material3.*
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.github.se.icebreakrr.model.profile.Gender
+import com.github.se.icebreakrr.model.profile.Profile
+import com.github.se.icebreakrr.model.profile.ProfilesViewModel
+import com.github.se.icebreakrr.model.tags.TagsViewModel
+import com.github.se.icebreakrr.ui.navigation.NavigationActions
+import com.github.se.icebreakrr.ui.navigation.TopLevelDestinations
+import com.github.se.icebreakrr.ui.profile.ProfilePictureSelector
+import com.github.se.icebreakrr.ui.tags.TagSelector
+import com.google.firebase.Firebase
+import com.google.firebase.Timestamp
+import com.google.firebase.auth.auth
+import com.google.firebase.messaging.FirebaseMessaging
+import java.util.Calendar
+
+@Composable
+fun ProfileCreationScreen(
+    tagsViewModel: TagsViewModel,
+    profilesViewModel: ProfilesViewModel,
+    navigationActions: NavigationActions
+) {
+
+  val configuration = LocalConfiguration.current
+  val screenWidth = configuration.screenWidthDp.dp
+  val screenHeight = configuration.screenHeightDp.dp
+
+  val profilePictureSize = screenWidth * 0.25f
+
+  val scrollState = rememberScrollState()
+  val fullName = remember { mutableStateOf("") }
+  val catchphrase = remember { mutableStateOf("") }
+  val description = remember { mutableStateOf("") }
+
+  val tempBitmap = profilesViewModel.tempProfilePictureBitmap.collectAsState().value
+  val user = profilesViewModel.selectedProfile.collectAsState().value
+
+  val selectedTags = tagsViewModel.filteringTags.collectAsState().value
+  val tagsSuggestions = tagsViewModel.tagsSuggestions.collectAsState()
+  val stringQuery = tagsViewModel.query.collectAsState()
+
+  val expanded = remember { mutableStateOf(false) }
+
+  var showDialog by remember { mutableStateOf(false) }
+  var isModified by remember { mutableStateOf(false) }
```

Comment:
You sometimes set this var to true, but never actually use it. Is this intentional ?

Reply:
No I will remove showDialog